src/array.vala

```diff
@@ -113,41 +113,7 @@ public class Vast.Array : Object
     public Value
     get_value ([CCode (array_length = false)] ssize_t[] index)
     {
-        var _value = Value (scalar_type);
-
-        if (scalar_type == typeof (string)) {
-            _value.set_string ((string) get_pointer (index));
-        }
-
-        else if (_value.fits_pointer ()) {
-            Memory.copy (_value.peek_pointer (), get_pointer (index), scalar_size);
-        }
-
-        else if (scalar_type == Type.BOXED) {
-            _value.set_boxed (get_pointer (index));
-        }
-
-        else if (scalar_type == typeof (char)) {
-            _value.set_char (*(char*) get_pointer (index));
-        }
-
-        else if (scalar_type == typeof (uint8)) {
-            _value.set_uchar (*(uchar*) get_pointer (index));
-        }
-
-        else if (scalar_type == typeof (int64)) {
-            _value.set_int64 (*(int64*) get_pointer (index));
-        }
-
-        else if (scalar_type == typeof (double)) {
-            _value.set_double (*((double*) get_pointer (index)));
-        }
-
-        else {
-            assert_not_reached ();
-        }
-
-        return _value;
+        return memory_to_value(_pointer_from_offset (_offset_from_index (index)));
```

Comment:
We could move the `_pointer_to_offset` into `memory_from_value` because even the iterator has to fetch the `_cached_data`. It could save a couple of indirection just to pass the offset from the iterator.